Packs/AWS-WAF/Integrations/AWS-WAF/AWS_WAF.py

```diff
@@ -0,0 +1,966 @@
+TEXT_TRANSFORMATIONS = 'NONE | COMPRESS_WHITE_SPACE | HTML_ENTITY_DECODE | LOWERCASE | CMD_LINE | URL_DECODE | ' \
+                       'BASE64_DECODE | HEX_DECODE | MD5 | REPLACE_COMMENTS | ESCAPE_SEQ_DECODE | SQL_HEX_DECODE | ' \
+                       'CSS_DECODE | JS_DECODE | NORMALIZE_PATH | NORMALIZE_PATH_WIN | REMOVE_NULLS | ' \
+                       'REPLACE_NULLS | BASE64_DECODE_EXT | URL_DECODE_UNI | UTF8_TO_UNICODE'
+
+''' HELPER FUNCTIONS '''
+
+
+def get_tags_dict_from_args(tag_keys: list, tag_values: list) -> List[dict]:
+    tags: list = []
+    if len(tag_keys) != len(tag_values):
+        raise DemistoException('Tha tags_keys and tag_values arguments must be at the same length.')
+
+    # keys and values are in the same length
+    n = len(tag_keys)
+    for i in range(n):
+        tag = {'Key': tag_keys[i], 'Value': tag_values[i]}
+        tags.append(tag)
+
+    return tags
+
+
+def build_regex_pattern_object(regex_patterns: list) -> List[dict]:
+    regex_patterns_objects: list = [
+        {'RegexString': regex_pattern} for regex_pattern in regex_patterns
+    ]
+    return regex_patterns_objects
+
+
+def build_visibility_config_object(metric_name: str,
+                                   cloud_watch_metrics_enabled: bool,
+                                   sampled_requests_enabled: bool) -> dict:
+    return {
+        'CloudWatchMetricsEnabled': cloud_watch_metrics_enabled,
+        'MetricName': metric_name,
+        'SampledRequestsEnabled': sampled_requests_enabled
+    }
+
+
+def build_ip_rule_object(args: dict) -> dict:
+    ip_rule: dict = {'Statement': {}}
+    ip_set_arn = argToList(args.get('ip_set_arn')) or []
+    condition_operator = args.get('condition_operator', '')
+    if len(ip_set_arn) > 1 and not condition_operator:
+        raise DemistoException('Please provide a value to the condition_operator argument '
+                               'when ip_set_arn has more than one value.')
+
+    if len(ip_set_arn) == 1:
+        ip_rule['Statement'] = build_ip_statement(ip_set_arn)
+    elif len(ip_set_arn) > 1:
+        statement_operator = OPERATOR_TO_STATEMENT_OPERATOR[condition_operator]
+        ip_rule['Statement'][statement_operator] = {
+            'Statements': build_ip_statement(ip_set_arn)
+        }
+    return ip_rule
+
+
+def build_ip_statement(ip_set_arn: list) -> list | dict:
+    ip_statements = [
+        {'IPSetReferenceStatement': {'ARN': ip_set}} for ip_set in ip_set_arn
+    ]
+    return ip_statements[0] if len(ip_statements) == 1 else ip_statements
+
+
+def build_country_statement(country_codes: list) -> dict:
+    return {'GeoMatchStatement': {'CountryCodes': country_codes}}
+
+
+def build_country_rule_object(args: dict) -> dict:
+    country_codes = argToList(args.get('country_codes')) or []
+    country_rule: dict = {
+        'Statement': build_country_statement(country_codes)
+    }
+    return country_rule
+
+
+def build_string_match_statement(args: dict) -> dict:
```

Comment:
suggestion:
you can pass **args
and in the definition of the func to do:
`def build_string_match_statement(match_type: str | None = '', string_to_match: str | None = '' ) -> dict:`
instead of doing `args.get` in the code

if is useful to use it when you assigning the same values in the name of the keys.